core/src/main/scala/org/apache/spark/util/AkkaUtils.scala

```diff
@@ -171,14 +215,20 @@ private[spark] object AkkaUtils extends Logging {
     while (attempts < maxAttempts) {
       attempts += 1
       try {
-        val future = actor.ask(message)(timeout)
-        val result = Await.result(future, timeout)
+        val future = actor.ask(message)(confTimeout.timeout)
+        val result = Await.result(future, confTimeout.timeout)
         if (result == null) {
           throw new SparkException("Actor returned null")
         }
         return result.asInstanceOf[T]
       } catch {
         case ie: InterruptedException => throw ie
+        case te: TimeoutException =>
+          var msg = te.toString()
+          if (confTimeout.description != null) {
+            msg += " with [" + confTimeout.description + "]"
```

Comment:
I'd like a little clearer explanation here, like "This timeout is controlled by spark.xxx.yyy.zzz"